core/state_processor.go

```diff
@@ -764,24 +773,57 @@ func RedeemLockedQuai(hc *HeaderChain, statedb *state.StateDB, currentBlockHeigh
 				lockup := params.LockupByteToBlockDepth[lockupByte]
 				if lockup == blockDepth {
 					balance := etx.Value()
+
+					if !statedb.Exist(internal) {
+						newAccountCreationGas := params.CallNewAccountGas(evm.Context.QuaiStateSize)
+
+						// Convert newAccountCreationGas (uint64) to *big.Int
+						newAccountCreationGasBigInt := new(big.Int).SetUint64(newAccountCreationGas)
+
+						// Check if balance is greater than or equal to newAccountCreationGasBigInt
+						if balance.Cmp(newAccountCreationGasBigInt) >= 0 {
```

Comment:
You can't compare gas and balance, they are different. One is a measure of compute/state units and the other is a currency

Reply:
You also can't multiply the block base fee by the gas to get the fee, because the miner of the block can arbitrarily increase the block base fee

Reply:
How would you suggest changing it?

Reply:
Where am I multiplying?

Reply:
This balance check is seeing if the balance is enough to cover the gas in order to make the account. In this case, they would be the same units unless I am mistaken.

Reply:
Gas units and balance are not the same thing